node/src/components/consensus/highway_core/state.rs

```diff
@@ -890,6 +890,22 @@ impl<C: Context> State<C> {
         result
     }
 
+    /// Drops all state other than evidence.
+    pub(crate) fn retain_evidence_only(&mut self) {
+        self.units.clear();
+        self.blocks.clear();
+        self.panorama = self
+            .panorama
+            .iter()
+            .map(|obs| match obs {
+                Observation::Faulty => Observation::Faulty,
+                Observation::None | Observation::Correct(_) => Observation::None,
+            })
+            .collect();
```

Comment:
We could just mutate this in place, right?

Reply:
Done in https://github.com/CasperLabs/casper-node/pull/979/commits/ed679ab721e1af26566b8ee7e9b6125d267e4c8f.